waitlist/utility/flask/__init__.py

```diff
@@ -131,23 +133,17 @@ def check_all_alts_authorized():
                      request.path, allowed_endpoints)
         return None
 
-    unauthed_chars: List[Character] = []
     # check all the chars are authorized
     # basically there is a sso token for all the alts
+    accountSSOTokenCharacterIDs = db.session.query(distinct(SSOToken.characterID)).\
+        filter(SSOToken.accountID == current_user.id)
 
-    # TODO: this could be done with an sql query
-    for character in user.characters:
-        found_key: bool = False
-        for sso_token in user. ssoTokens:
-            if sso_token.characterID == character.id:
-                found_key = True
-                break
-
-        if not found_key:
-            unauthed_chars.append(character)
-
+    unauthed_chars: List[Character] = db.session.query(Character).\
+        join(Account.characters).\
+        filter(Account.id == current_user.id, Character.id.notin_(accountSSOTokenCharacterIDs)).\
+        all()
     # we found at least 1 not authed character
-    if len(unauthed_chars) > 0:
+    if len(unauthed_chars) > 0:       
```

Comment:
You added some whitespace here should be removed:
```suggestion
    if len(unauthed_chars) > 0:
```